src/HomeAssistantManager.cpp

```
@@ -188,8 +188,11 @@ bool HomeAssistantManager::begin() {
 
 bool HomeAssistantManager::enqueuePublish(const HAPublishRequest& req) {
     if (publishQueue == nullptr) return false;
-    // Overwrite oldest if full (drop semantics for state-based retained messages)
-    return xQueueSend(publishQueue, &req, 0) == pdTRUE;
+    if (xQueueSend(publishQueue, &req, 0) != pdTRUE) {
```

Cppcheck (2.20.0)

[style] The function 'print' is never used.

(unusedFunction)